File: Data Structures & Algorithms/non-overlapping-intervals/submission-1.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>& b){
            if(a[1]==b[1]) return a[0]<b[0];
            return a[1]<b[1];
        });
     
        int i=0, cnt=0;
        while(i<intervals.size()) {
            int j=i;
            while(i<intervals.size() && intervals[i][0]<intervals[j][1]) i++;
            cnt++;
        }
        return intervals.size()-cnt;
    }
};

File: Data Structures & Algorithms/meeting-rooms-iii/submission-1.java
class Solution {
    public int mostBooked(int n, int[][] meetings) {
        Arrays.sort(meetings, (a, b) -> a[0]-b[0]);
        int[] rooms = new int[n];
        Queue<Integer> qu = new PriorityQueue<>();
        Queue<int[]> resv = new PriorityQueue<>((a,b) -> a[0]-b[0]);
        for(int i=0;i<n;i++) {
            qu.add(i);
        }
        int mx=0, st=0;
        for(int[] meet: meetings) {
            st = Math.max(meet[0], st);
            while(!resv.isEmpty()&&resv.peek()[0]<=st) {
                int[] unreserve = resv.remove();
                qu.add(unreserve[1]);
            }
            if(qu.isEmpty()) {
                st = Math.max(resv.peek()[0], st);
                while(!resv.isEmpty()&&resv.peek()[0]<=st) {
                    int[] unreserve = resv.remove();
                    qu.add(unreserve[1]);
                }
            }
            int reserve = qu.remove();
            // System.out.println(""+reserve+" "+meet[0]+" "+st);
            rooms[reserve]++;
            mx = Math.max(rooms[reserve], mx);
            resv.add(new int[]{meet[1]+st-meet[0], reserve});
        }
        int cnt=0;
        for(int i=0;i<rooms.length;i++) {
            if(mx==rooms[i]) return i;
        }
        return cnt;
    }
}